packages/comms/utils/index.ts

```diff
@@ -0,0 +1,253 @@
+            if (err) reject(err);
+            resolve(client);
+        });
+    }).then(client => {
+        const wsdlDescr = client.describe();
+        return [client.wsdl, wsdlDescr];
+    });
+}
+
+function printUsage() {
+    console.log("Usage: npm run wsdl-to-ts -- --url=someUrl\n");
+    console.log("Available flags: ");
+    console.log("====================");
+    console.log("--url=someUrl\t\t\tA URL for a WSDL to be converted to TypeScript interfaces");
+    console.log("--outDir=./some/path\t\tThe directory into which the generated TS interfaces will be written (defaults to \"./wsdl\").");
+    console.log("--print\t\t\t\tRather than writing files, print the generated TS interfaces to the CLI");
+}
+
+if (!args.url) {
+    console.error("No WSDL Url provided.\n");
+    printUsage();
+    process.exit(0);
+}
+
+if (args.help) {
+    printUsage();
+    process.exit(0);
+}
+
+function parseEnum(enumString: string, enumEl) {
+    const enumParts = enumString.split("|");
+    printDbg(`parsing enum parts ${enumParts[0]}`, enumParts);
+    return {
+        type: enumParts[0],
+        enumType: enumParts[1].replace(/xsd:/, ""),
+        values: enumParts[2].split(",").map((v, idx) => {
+            const member = v.split(" ").map(w => changeCase(w, Case.PascalCase)).join("");
+            if (enumParts[1].replace(/xsd:/, "") === "int") {
+                let memberName = "";
+                enumEl.children.filter(el => el.name === "annotation")[0].children.forEach(el => {
+                    memberName = changeCase(el.children[idx].$description, Case.PascalCase).replace(/ /g, "");
+                });
+                return `${memberName} = ${member}`;
+            }
+            return `${member} = "${member}"`;
+        })
+    };
+}
+
+function parseTypeDefinition(operation: JsonObj, opName: string, types) {
+
+    const typeDefn: JsonObj = {};
+    printDbg(`processing ${opName}`, operation);
+    for (const prop in operation) {
+        const propName = (prop.indexOf("[]") < 0) ? prop : prop.slice(0, -2);
+        if (typeof operation[prop] === "object") {
+            const op = operation[prop];
+            if (knownTypes.indexOf(propName) < 0) {
+                knownTypes.push(propName);
+                const defn = parseTypeDefinition(op, propName, types);
+                if (prop.indexOf("[]") > -1) {
+                    typeDefn[propName] = prop;
+                } else {
+                    typeDefn[propName] = defn;
+                }
+                parsedTypes[propName] = defn;
+            } else {
+                typeDefn[propName] = prop;
+            }
+
+        } else {
+            if (ignoredWords.indexOf(prop) < 0) {
+                const primitiveType = operation[prop].replace(/xsd:/gi, "");
+                if (prop.indexOf("[]") > 0) {
+                    typeDefn[prop.slice(0, -2)] = primitiveType + "[]";
+                } else if (operation[prop].match(/.*\|.*\|.*/)) {
```

Comment:
I have no idea what this looks like, but does this catch enums with only 2 options (TRUE|FALSE)?

Reply:
Added a comment to this. Looking at it again, realized that regex wasn't matching value(s) of an enum, but the string that node soap generates to represent an enum, which is always EnumName|type|values